test/integration/vitesscluster/topology_test.go

```diff
@@ -0,0 +1,233 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vitesscluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/topo"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/test/integration/framework"
+)
+
+func TestTopology(t *testing.T) {
+	ctx := context.Background()
+
+	f := framework.NewFixture(ctx, t)
+	defer f.TearDown()
+
+	ns := "default"
+	cluster := "test-topology"
+
+	// Start with the same basic VitessCluster as the regular test.
+	vt := &planetscalev2.VitessCluster{}
+	framework.MustDecodeYAML(basicVitessCluster, vt)
+
+	// Set the cluster to use an external lockserver, namely the same etcd that
+	// we use for our test k8s apiserver.
+	//
+	// This lets us actually test the operator's topology interaction.
+	// Otherwise, there is no actual etcd since this integration test
+	// environment doesn't actually run any Pods.
+	params := &planetscalev2.VitessLockserverParams{
+		Implementation: "etcd2",
+		Address:        framework.EtcdURL(),
+		RootPath:       fmt.Sprintf("/vitess/%s/", cluster),
+	}
+	vt.Spec.GlobalLockserver.External = params
+
+	f.CreateVitessCluster(ns, cluster, vt)
+
+	ts, err := topo.OpenServer(params.Implementation, params.Address, params.RootPath)
+	if err != nil {
+
+	}
+
+	// Look for topology entries that are populated by the operator.
+	verifyTopoRegistration(f, ts)
+
+	// Create some orphaned topology entries and check that the operator prunes them.
+	verifyTopoPruning(f, ts)
+}
+
+func verifyTopoRegistration(f *framework.Fixture, ts *topo.Server) {
+	f.WaitFor("Cells Alias to be registered", func() error {
+		aliases, err := ts.GetCellsAliases(f.Context(), true)
+		if err != nil {
+			return err
+		}
+		if aliases["planetscale_operator_default"] == nil {
+			return fmt.Errorf("planetscale_operator_default cells alias not found")
+		}
+		return nil
+	})
+
+	f.WaitFor("Cells to be registered", func() error {
+		if _, err := ts.GetCellInfo(f.Context(), "cell1", true); err != nil {
+			return err
+		}
+		if _, err := ts.GetCellInfo(f.Context(), "cell2", true); err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+func populateTopo(f *framework.Fixture, ts *topo.Server) {
+	// Add an orphan SrvKeyspace.
+	if err := ts.UpdateSrvKeyspace(f.Context(), "cell1", "delete_me", &topodatapb.SrvKeyspace{}); err != nil {
+		f.Fatalf("Can't create SrvKeyspace: %v", err)
+	}
+
+	// Add an orphan Cell.
```

Comment:
Same here. Is `Cell` intentionally capitalized? 
```suggestion
	// Add an orphan cell.
```

Reply:
Yeah it's intentional. It's sort of carrying over the documentation habit of capitalizing Official API Concepts into the Vitess world.